src/ORM/Association/BelongsToMany.php

```diff
@@ -1278,12 +1256,26 @@ protected function _buildQuery($options)
 
         // Ensure that association conditions are applied
         // and that the required keys are in the selected columns.
+
+        $tempName = $this->_name . '_CJoin';
+        $schema = $assoc->schema();
```

Comment:
We directly/indirectly call `$assoc->target()` ~3 times in this method. Should we save some function calls by doing `$target = $assoc->target()`?